src/sb-util.ts

```diff
@@ -39,23 +48,35 @@ The classes below make up the sb-util API.
 const storage = new WeakMap();
 
 export class ScratchProject {
-    public constructor(projectJSON: JSON) {
-        storage.set(this, projectJSON);
+    /**
+     * The project JSON as read from the project.
+     */
+    public projectJSON: SB3ProjectJSON;
+    /**
+     * An interface capable of fetching the assets, later used by AssetCollections.
+     */
+    public assetFetcher: AssetFetcher;
+
+    public constructor(projectJSON: SB3ProjectJSON | JSON, assetFetcher: AssetFetcher = NoAssetFetcher) {
+        this.projectJSON = projectJSON as SB3ProjectJSON;
+        this.assetFetcher = assetFetcher;
     }
 
     // DISABLING ESLINT: a prop can be a string, number, object, or boolean
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     public prop(property: string): any {
-        return storage.get(this)[property];
+        return this.projectJSON[property];
     }
 
     public sprites(selector?: string): SpriteCollection {
         if (selector && typeof selector !== 'string')
             throw new Error('SpriteCollection selector should be a string!');
 
-        const sprites: Iterable<SpriteProperties> = this.prop(ScratchProjectKeys.TARGETS);
-        if (!selector && typeof selector !== 'string') return new SpriteCollection(sprites);
-        return new SpriteCollection(sprites).query(selector);
+        const collection = new SpriteCollection(this.projectJSON.targets);
+        if (selector) {
```

Comment:
I think we'll see an idiom around this that would benefit from some kind of helper but until then ...